vendor/bat-native-ledger/src/bat/ledger/internal/grants.cc

```diff
@@ -79,12 +91,14 @@ void Grants::GetGrantsCallback(
                                                                   &statusCode,
                                                                   &error);
   if (hasResponseError && statusCode == net::HTTP_NOT_FOUND) {
+    callback(ledger::Result::GRANT_NOT_FOUND);
     ledger_->SetLastGrantLoadTimestamp(time(0));
     ledger_->OnGrant(ledger::Result::GRANT_NOT_FOUND, properties);
```

Comment:
this shouldn't be needed anymore, we should remove `ledger_client_->OnGrant`

Reply:
so `LedgerImpl::FetchGrants` should add callback to `LedgerImpl::OnGrant` and instead of `ledger_client_->OnGrant` we should do `callback`